File: algorithms/algorithm1.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char *argv[]) {     
    int n;
    sscanf (argv[1],"%d",&n);
    double pi = 0;
    int sign = 1;
    int i;
    for(i = 1; i<=n; i=i+2){
        pi = pi + (double)sign*1/i;
        sign = sign * -1;
    }
    pi = pi*4;
    printf("%.30f",pi);
    return 0;
}
    

File: algorithms/algorithm1.py
import sys
def main(argv):
    n = int(argv[0])
    pi = 0
    sign = 1
    for i in range(1,n,2):
        pi = pi + sign*1/i
        sign = sign * -1
    pi = pi*4
    print(pi, end="")

if __name__ == "__main__":
   main(sys.argv[1:])
